tools/crm_resource.c

```diff
@@ -1844,17 +1875,11 @@ main(int argc, char **argv)
             break;
 
         case cmd_why:
-            {
-                pe_node_t *dest = NULL;
-
-                if (options.host_uname) {
-                    dest = pe_find_node(data_set->nodes, options.host_uname);
-                    if (dest == NULL) {
-                        rc = pcmk_rc_node_unknown;
-                        goto done;
-                    }
-                }
-                out->message(out, "resource-reasons-list", cib_conn, data_set->resources, rsc, dest);
+            if ((options.host_uname != NULL) && (node == NULL)) {
+                rc = pcmk_rc_node_unknown;
+            } else {
+                out->message(out, "resource-reasons-list", cib_conn,
+                             data_set->resources, rsc, node);
```

Comment:
While you're here, rc should probably be assigned the return value of out->message.  This particular message is always returning pcmk_rc_ok for the moment, but that could potentially change in the future.

Reply:
Good point